tidy3d/components/data/monitor_data.py

```diff
@@ -575,6 +578,143 @@ class FluxTimeData(MonitorData):
+            src_amps = source_spectrum_fn(field_data.f)
+            fields_norm[field_name] = field_data / src_amps
+
+        return self.copy(update=fields_norm)
+
+    def _make_coords(self):
+        """Helper to make the coordinates dictionary."""
+        coords = {}
+        coords["orders_x"] = np.atleast_1d(self.E.orders_x.values)
+        coords["orders_y"] = np.atleast_1d(self.E.orders_y.values)
+        coords["polarization"] = ["s", "p"]
+        coords["f"] = np.array(self.E.f.values)
+        return coords
+
+    @property
+    def angles(self):
+        """Return the (theta, phi) angles corresponding to each allowed pair of diffraction
+        orders as a data array. Disallowed angles are set to ``np.nan``.
+        """
+        # some wave number pairs are outside the light cone, leading to warnings from numpy.arcsin
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore", message="invalid value encountered in arcsin", category=RuntimeWarning
+            )
+            thetas, phis = DiffractionMonitor.kspace_2_sph(
+                np.array(self.ux)[:, None, :], np.array(self.uy)[None, :, :], axis=2
+            )
+
+        coords = self._make_coords()
+        del coords["polarization"]
+        theta_data = xr.DataArray(thetas, coords=coords)
+        phi_data = xr.DataArray(phis, coords=coords)
+        return theta_data, phi_data
+
+    @property
+    def amps(self):
+        """Return normalized field amplitudes in each order, for 's' and 'p' polarizations.
+        Fields are normalized so that the power equals abs(amps)^2 for each polarization.
+        """
+        e_x = self.E.sel(polarization="x").values
+        e_y = self.E.sel(polarization="y").values
+
+        e_theta = np.zeros_like(e_x)
+        e_phi = np.zeros_like(e_x)
+
+        thetas, phis = self.angles
+
+        # compute normalized fields in local spherical coordinates
+        # the terms of the normalization factor are:
+        # 2: this comes from the time-averaged Poynting vector definition
+        # 'ETA_0': this relates E and H components
+        norm = 1.0 / np.sqrt(2.0 * ETA_0)
+        e_phi = np.nan_to_num(-np.sin(phis) * e_x + np.cos(phis) * e_y) * norm
+        e_theta = np.nan_to_num((e_x * np.cos(phis) + e_y * np.sin(phis)) / np.cos(thetas)) * norm
+
+        # stack the fields in s- and p-components along a new polarization axis
+        return DiffractionDataArray(np.stack([e_phi, e_theta], axis=2), coords=self._make_coords())
+
+    @property
+    def power(self):
+        """Return the total power in each order, summed over both polarizations."""
+        amps = self.amps
+        angles = self.angles
+        cos_theta = np.cos(np.nan_to_num(angles[0]))
+
+        amps_s = amps.sel(polarization="s")
+        amps_p = amps.sel(polarization="p")
+        return (np.abs(amps_s) ** 2 + np.abs(amps_p) ** 2) * cos_theta
+
```

Comment:
nit: add output type signature to all of those; also, `abs(amps)^2` in the docstring should probably better be ```
```
``abs(amps)**2``
```